src/metabase/task/search_index.clj

```diff
@@ -33,7 +33,10 @@
 
 ;; We define the job bodies outside the defrecord, so that we can redefine them live from the REPL
 
-(defn- report->prometheus! [report]
+(defn- report->prometheus! [timer report]
+  (let [total-cnt (reduce + (map second report))]
+    (prometheus/observe! :metabase-search/index-ms (/ (u/since-ms timer)
```

Comment:
I'd think we're more interested in the total time per batch?

We can always divide by the number of entries to get the average per entry on the other side. We wouldn't care that we're aggregating over multiple batches potentially there.

Reply:
It's going to be way more variable depending on a batch size and bucketing will be completely broken.

Reply:
If you want proper bucking with these averages though, you'd need to submit N observations as well?

I just don't see how this is very useful - you cant use these values in any aggregations. 

There's also quite a bit of overhead for doing a batch, so the time won't be linear and there will still be correlation with the batch size.